MovingBlocks/Terasology
# build(facades.PC): tidy distribution tasks

This is a follow-up to #4343. I'm not aware of any open issues fixed by this, but it makes some build dependencies more consistent with the way #4343 manages them and addresses some TODOs that have been in the gradle files for a while.

- Removes `RemoteModuleGatherer`, the code that puts **`.jar` files in `modules/`** during gradle dependency resolution. It's unnecessary under the modules-on-the-classpath strategy used by #4343, figuring out a better way to provide those dependencies without breaking the sandbox is #4375.
- Removes PC facade targets `distApp` and `distModules`, `distPCZip`, they are unused.
  - the gradle-standard `distTar` and `distZip` tasks are supported.
- Keeps `distForLauncher` so we don't need to change any of Launcher's assumptions about the zip file layout.
- Removes some overrides to gradle/IntelliJ build directories that were counterproductive.


### Caveat

If you build a dist while you have module subprojects present, they will get lumped in on the classpath and not separated.

I think that's not a showstopper because our release builds are made without modules and then a separate process adds them to the distribution later.


### How to test

- make some zips using `:facades:PC:distZip` and `distForLauncher`; make sure they are something you can unzip and start up (outside your development workspace) using the script included in its `bin/`
- in-game look at Extras / Credits and make sure it displays the credits file.


### Outstanding before merging

- [ ] merge gradle upgrade #4439 
- [ ] do the start scripts need to be customized? (the only arg the old start scripts added was `-Xmx`)
